source/server/listener_impl.cc

```diff
@@ -263,6 +263,8 @@ ListenerImpl::ListenerImpl(const envoy::config::listener::v3::Listener& config,
       listener_filters_timeout_(
           PROTOBUF_GET_MS_OR_DEFAULT(config, listener_filters_timeout, 15000)),
       continue_on_listener_filters_timeout_(config.continue_on_listener_filters_timeout()),
+      disable_tls_inspector_injection_(Runtime::runtimeFeatureEnabled(
```

Comment:
why do we need this variable and the change involves multiple places? Can we just check the runtime feature in `needTlsInspector` and return false if the runtime flag is true?

Reply:
Sorry, for some reason I'd thought this should be checked on class initialization